pkg/log/logger.go

```diff
@@ -0,0 +1,295 @@
+// Copyright 2019-2020 Intel Corporation. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package log
+
+import (
+	"fmt"
+	"math"
+	"os"
+)
+
+// Level describes the severity of log messages.
+type Level int
+
+const (
+	// LevelDebug is the severity for debug messages.
+	LevelDebug Level = iota
+	// LevelInfo is the severity for informational messages.
+	LevelInfo
+	// LevelWarn is the severity for warnings.
+	LevelWarn
+	// LevelError is the severity for errors.
+	LevelError
+	// LevelFatal is the severity for fatal errors.
+	LevelFatal
+	// LevelPanic is the severity for panic messages.
+	LevelPanic
```

Comment:
At least it was not my intention. Do you ask that because a panic() could be recover()ed form in some situations but you can't prevent os.Exit() from happening ? If you do, then I see your point and, yes, we should probably do that.

Reply:
Yes, that was my thinking. Panic can probably be recovered (just a bug in some functionality) but Fatal cannot

Reply:
Swapped LevelFatal and LevelPanic.